frontend/routes/Wifi.jsx

```diff
@@ -1,53 +1,99 @@
-import * as React from 'react';
+import React, { useEffect, useState } from 'react';
+import { WifiDetails } from '../components/WifiDetails';
+import Typography from '@material-ui/core/Typography';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import Skeleton from '@material-ui/lab/Skeleton';
+import { makeStyles } from '@material-ui/core/styles';
+
 import { exec_table, exec } from '../util/exec';
+import { sortArrayOfObjects } from '../util/sort';
 
 export const title = 'Wifi Configuration';
 export const route = '/wifi';
 
-export class Component extends React.Component {
-  constructor(props) {
-    super(props);
+const useStyles = makeStyles(theme => ({
+  wifiNetwork: {
+    margin: '2em',
+    marginTop: '30px',
+    width: '80%',
+    margin: '0 auto',
+  },
+  header: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '20px',
+    [theme.breakpoints.up('md')]: {
+      flexDirection: 'row',
+      alignItems: 'center',
+      justifyContent: 'space-between',
+    },
+  },
+  select: {
+    width: '220px',
+  },
+  skeletonWrapper: {
+    margin: '20px 0',
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '20px',
+  },
+}));
+
+export function Component(props) {
+  const classes = useStyles();
+  const [wifi_networks, setWifi_networks] = useState();
+  const [last_refresh, setLast_refresh] = useState();
+  const [sortBy, setSortBy] = useState('Default');
 
-    this.state = {
-      wifi_networks: [],
-      last_refresh: '',
-    };
+  useEffect(() => {
     setInterval(() => {
-      exec_table('nmcli dev wifi').then(result => this.setState({ wifi_networks: result }));
-      exec('date').then(result => this.setState({ last_refresh: result }));
+      exec_table('nmcli dev wifi').then(result => setWifi_networks(result));
+      exec('date').then(result => setLast_refresh(result));
     }, 1000);
-  }
+  }, []);
 
-  render() {
-    return (
-      <div>
-        <Table data={this.state.wifi_networks} />
-      </div>
-    );
-  }
-}
+  const handleChange = event => {
+    setSortBy(event.target.value);
+    setWifi_networks(sortArrayOfObjects(wifi_networks, event.target.value));
```

Comment:
This only sorts the array once, it needs to be sorted everytime new wifi data is received.